Libraries/xcdriver/Sources/HelpAction.cpp

```diff
@@ -0,0 +1,221 @@
+/**
+ Copyright (c) 2016-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <xcdriver/HelpAction.h>
+#include <xcdriver/UsageAction.h>
+#include <libutil/FSUtil.h>
+#include <libutil/SysUtil.h>
+
+using xcdriver::HelpAction;
+using libutil::FSUtil;
+using libutil::SysUtil;
+
+HelpAction::
+HelpAction()
+{
+}
+
+HelpAction::
+~HelpAction()
+{
+}
+
+int HelpAction::
+Run()
+{
+    UsageAction::Run();
```

Comment:
Interesting. I would probably structure this by pulling the usage code out into a separate class (that just returns the usage text as a string), then making both this action and the `HelpAction` print it. That way the `Action` classes have just the responsibility of performing the action from the command line, but not implementing the underlying functionality.